archunit/src/test/java/com/tngtech/archunit/library/plantuml/rules/PlantUmlParserTest.java

```diff
@@ -418,6 +406,54 @@ public void parses_a_component_diagram_that_uses_alias_with_and_without_brackets
         assertThat(bar.getDependencies()).isEmpty();
     }
 
+    @Test
+    public void ignore_components_that_are_not_yet_defined() {
+        File file = TestDiagram.in(temporaryFolder)
+                .dependencyFrom("[NotYetDefined]").to("[AlsoNotYetDefined]")
+                .write();
+
+        PlantUmlDiagram diagram = createDiagram(file);
+
+        assertThat(diagram.getComponentsWithAlias()).isEmpty();
+    }
+
+    @Test
+    public void ignore_database_components() {
+        File file = TestDiagram.in(temporaryFolder)
+                .rawLine("database \"DB\"")
+                .component("componentA").withAlias("aliasA").withStereoTypes("..packageA..")
+                .component("componentB").withAlias("aliasB").withStereoTypes("..packageB..")
+                .dependencyFrom("aliasA").to("aliasB")
+                .dependencyFrom("aliasB").to("DB")
+                .dependencyFrom("componentA").to("DB")
+                .write();
+
+        PlantUmlDiagram diagram = createDiagram(file);
+
+        PlantUmlComponent a = getComponentWithAlias(new Alias("aliasA"), diagram);
+        PlantUmlComponent b = getComponentWithAlias(new Alias("aliasB"), diagram);
+
+        assertThat(a.getDependencies()).containsOnly(b);
+        assertThat(b.getDependencies()).isEmpty();
+    }
+
+    @Test
+    public void components_with_component_keyword() {
```

Comment:
```suggestion
    public void supports_components_declared_with_component_keyword() {
```

Reply:
fixed and moved to separated PR
commit: https://github.com/TNG/ArchUnit/pull/1223/commits/39190ba798e524f8bd57472e717a0f0653a3c4f3
PR: (https://github.com/TNG/ArchUnit/pull/1223)